client/app/components/form/MultiStepButtons.tsx

```diff
@@ -6,26 +6,29 @@ import Link from "next/link";
 
 interface SubmitButtonProps {
   baseUrl: string;
-  disabled?: boolean;
   cancelUrl: string;
   classNames?: string;
+  disabled?: boolean;
+  isSubmitting: boolean;
   step: number;
   steps: string[];
   allowBackNavigation?: boolean;
 }
 
 const SubmitButton: React.FunctionComponent<SubmitButtonProps> = ({
+  allowBackNavigation,
   baseUrl,
+  cancelUrl,
+  classNames,
   disabled,
+  isSubmitting,
   step,
   steps,
-  cancelUrl,
-  classNames,
-  allowBackNavigation,
 }) => {
   const { pending } = useFormStatus();
   const isFinalStep = step === steps.length - 1;
-  const isDisabled = disabled || pending;
+  const isDisabled = disabled || pending || isSubmitting;
```

Comment:
This was a really good catch and sent me down quite the rabbit hole this morning trying to figure out why the `useFormStatus` hook wasn't firing. If it did work I think it would be much cleaner and preferable to use though we probably wouldn't have needed this ticket if it was working.

From what I can tell from the React docs you linked as well as the [NextJS Server Actions docs](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) this doesn't work with a `form` `onSubmit` and needs `<form action={handleAction}>` to work. 

With NextJS the hook seems to read a form status using server actions via the action prop. I tried in `<FormBase>` component but it doesn't even seem like RJSF supports using `action` so I'm not sure if we could even refactor it.

Anyways of course it's possible I missed something but for now I've remove `useFromState` from `MultiStepButtons` as well as `SubmitButtons` components as it wasn't firing. I did add a useEffect to multiple components to see if it was doing anything and it was always `false`.

<img width="254" alt="Screenshot 2024-01-17 at 8 26 52 AM" src="https://github.com/bcgov/cas-registration/assets/14259474/4b6f0866-29f7-479c-b75d-9fad06af267e">

Anyways, thanks I learned something today. Let me know if you have any other ideas if not we can keep using `isSubmitting`.